### PR TITLE
documentation metadata API

### PR DESCRIPTION
**Pull request by jchristgit**
This PR implements the API endpoints to accompany https://github.com/discord-python/bot/pull/53.
In a nutshell, it

- adds a new database "schema" for storing documentation metadata
- adds api endpoints for modifying this metadata
- adds tests for the above.

This way, documentation metadata used for the documentation lookup cog will be stored in the site's database and can easily be changed through (to be implemented-) bot commands.

### CODE REVIEW

File: pysite/views/api/bot/doc.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+        - If `package` is not provided, return all
+        packages known to the database.
+
+        Data must be provided as params.
+        API key must be provided as header.
+        """
+
+        if params:
+            package = params[0].get("package")
```
Comment (gdude2002) [inline pysite/views/api/bot/doc.py]: This schema setup is intended to allow one to specify multiple sets of parameters. You should support this in your endpoints.

File: pysite/views/api/bot/doc.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        Adds a new documentation metadata object.
+
+        If the `package` passed in the data
+        already exists, it will be updated instead.
+
+        Data must be provided as JSON.
+        API key must be provided as header.
+        """
+
+        json_data = json_data[0]
```
Comment (gdude2002) [inline pysite/views/api/bot/doc.py]: This schema setup is intended to allow one to specify multiple sets of parameters. You should support this in your endpoints.

File: pysite/views/api/bot/doc.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        """
+        Deletes a documentation metadata object.
+        Expects the `package` to be deleted to
+        be specified as a request parameter.
+
+        Data must be provided as params.
+        API key must be provided as header.
+        """
+
+        package = json_data[0]["package"]
```
Comment (gdude2002) [inline pysite/views/api/bot/doc.py]: This schema setup is intended to allow one to specify multiple sets of parameters. You should support this in your endpoints.

**Review by gdude2002 — CHANGES_REQUESTED** [review]

File: pysite/views/api/bot/doc.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        API key must be provided as header.
+        """
+
+        packages_to_insert = (
+            {
+                "package": json_object["package"],
+                "base_url": json_object["base_url"],
+                "inventory_url": json_object["inventory_url"]
+            }
+            for json_object in json_data
```
Comment (gdude2002) [inline pysite/views/api/bot/doc.py]: I think I'd prefer this line to be moved up, after the `}`

**Review by gdude2002 — DISMISSED** [review]

**Review by lemonsaurus — DISMISSED** [review]
This is just a really excellent PR.

**Review by gdude2002 — DISMISSED** [review]

**Review by gdude2002 — APPROVED** [review]

**Pull request merged**